packages/svelte/README.md

```diff
@@ -7,124 +7,333 @@
+npm install -D @carbon/charts-svelte d3@5.x
 ```
 
 If you prefer [Yarn](https://yarnpkg.com/en/), use the following command
 instead:
 
 ```bash
-yarn add @carbon/charts @carbon/charts-svelte d3@5.x
+yarn add -D @carbon/charts-svelte d3@5.x
 ```
 
-**Note:** you'd also need to install `carbon-components` if you're not using a bundled version of the library.
+**Note:** you'd also need to install `carbon-components` if you're not using a
+bundled version of the library.
 
-## Step-by-step instructions
+## Set-up
 
-Quickly scaffold a project from the official [Svelte webpack template](https://github.com/sveltejs/template-webpack) using [degit](https://github.com/Rich-Harris/degit):
+This is an overview of using Carbon Charts with common Svelte set-ups.
 
-```bash
-npx degit sveltejs/template-webpack svelte-app
-cd svelte-app
+-   [sveltekit](#sveltekit)
+-   [vite](#vite)
+-   [sapper](#sapper)
+-   [rollup](#rollup)
+-   [webpack](#webpack)
+-   [snowpack](#snowpack)
+
+### SvelteKit
+
+[SvelteKit](https://github.com/sveltejs/kit) is fast becoming the de facto
+framework for building Svelte apps that supports both client-side rendering
+(CSR) and server-side rendering (SSR).
+
+For set-ups powered by [vite](https://github.com/vitejs/vite), add
+`@carbon/charts` to the list of dependencies for `vite` to optimize.
+
+```js
+// svelte.config.js
+const config = {
+	kit: {
+		target: '#svelte',
+		vite: {
+			optimizeDeps: { include: ['@carbon/charts'] },
+		},
+	},
+};
+
+export default config;
 ```
 
-Install the dependencies:
+### Vite
 
-```bash
-yarn add carbon-components @carbon/charts @carbon/charts-svelte d3
+[vite-plugin-svelte](https://github.com/sveltejs/vite-plugin-svelte) is an
+alternative to using SvelteKit. Similarly, instruct `vite` to optimize
+`@carbon/charts` in vite.config.js.
+
+Note that `@sveltejs/vite-plugin-svelte` is the official vite/svelte integration
+not to be confused with [svite](https://github.com/svitejs/svite).
+
+```js
+// vite.config.js
+import { svelte } from '@sveltejs/vite-plugin-svelte';
+import { defineConfig } from 'vite';
+
+export default defineConfig(({ mode }) => {
+	return {
+		plugins: [svelte()],
+		build: { minify: mode === 'production' },
+		optimizeDeps: { include: ['@carbon/charts'] },
+	};
+});
 ```
 
-Let's add a simple bar chart from `@carbon/charts-svelte`.
+### Sapper
+
+[sapper](https://github.com/sveltejs/sapper) is another official Svelte
+framework that supports server-side rendering (SSR).
+
+Take care to install `@carbon/charts-svelte` as a development dependency.
+
+No additional configuration should be necessary.
+
+### Rollup
+
+When using this library with [rollup](https://github.com/rollup/rollup), set
+`context` to `"window"` in the exported configuration in `rollup.config.js`.
+
+```js
+// rollup.config.js
+
+export default {
+	context: 'window',
+	// ...
+};
+```
+
+#### Troubleshooting
+
+##### "process is not defined"
```

Comment:
this is probably irrelevant now since our rollup config does the same

Reply:
Unfortunately, I still encounter this error when using version ^0.41.90. Would you like me to record this as a separate issue?

Reply:
Are you sure? I don't see any references to `process.` in https://unpkg.com/@carbon/charts@0.41.90/bundle.js

Reply:
Double checked. With a complete fresh install, the resolved deps are:

- @carbon/charts-svelte@^0.41.90
- @carbon/charts@^0.41.90
- carbon-components@10.5.0

The IIFE bundle generated by *Rollup* still pulls in the `warning.js` from `warning@3.0.0`:

<img width="728" alt="Screen Shot 2021-07-21 at 12 57 21 PM" src="https://user-images.githubusercontent.com/10718366/126552219-a05d78d8-9242-4ead-b906-804efc17b3d1.png">

Reply:
oh ok, that'd be related to your svelte app I think, the charts bundle doesn't include references to `process`. Where is this piece of code coming from?

Reply:
Exactly. It's coming from `/node_modules/warning/browser.js`